hyperapp.d.ts

```diff
@@ -51,58 +56,126 @@ export function h<Props>(
 
 /** @namespace [App] */
 
-/** The result of an action.
- *
- * @memberOf [App]
+/** 
+ * Type to describe a nested map of type T
+ * 
+ * example: { 'foo' : { 'bar': 1 }, 'baz': '' }
  */
-export type ActionResult<State> = Partial<State> | Promise<any> | null | void
+type NestedMap<T> = {
+  [key: string]: NestedMap<T> | T
+}
 
-/** The interface for a single action implementation.
- *
- * @memberOf [App]
+/** 
+ * Type alias to proxy UnwiredActions to WiredActions.
  */
-export type ActionType<State, Actions> = (
-  data?: any
-) =>
-  | ((state: State, actions: Actions) => ActionResult<State>)
-  | ActionResult<State>
+type ActionProxy<
```

Comment:
Nice way to define the wiring. This was causing me some trouble understanding this out of the documentation.